python/generateTemplate.py

```diff
@@ -0,0 +1,1155 @@
+            for the mid-point of the next sub-band.
+            Otherwise yields a tuple of the x and y
+            DCR offsets for the start and end of the next sub-band.
+
+        """
+        zenith_angle = Angle(np.pi/2) - elevation
+        wavelength_midpoint = bandpass.calc_eff_wavelen()
+        delta = namedtuple("delta", ["start", "end"])
+        dcr = namedtuple("dcr", ["dx", "dy"])
+        if use_midpoint:
+            for wl in GenerateTemplate._wavelength_iterator(bandpass, use_midpoint=True):
+                # Note that refract_amp can be negative, since it's relative to the midpoint of the full band
+                refract_mid = diff_refraction(wavelength=wl, wavelength_ref=wavelength_midpoint,
+                                              zenith_angle=zenith_angle,
+                                              observatory=observatory, weather=weather)
+                refract_mid_pixels = refract_mid.asArcseconds()/pixel_scale.asArcseconds()
+                yield dcr(dx=refract_mid_pixels*np.sin(rotation_angle.asRadians()),
+                          dy=refract_mid_pixels*np.cos(rotation_angle.asRadians()))
+        else:
+            for wl_start, wl_end in GenerateTemplate._wavelength_iterator(bandpass, use_midpoint=False):
+                # Note that refract_amp can be negative, since it's relative to the midpoint of the full band
+                refract_start = diff_refraction(wavelength=wl_start, wavelength_ref=wavelength_midpoint,
+                                                zenith_angle=zenith_angle,
+                                                observatory=observatory, weather=weather)
+                refract_end = diff_refraction(wavelength=wl_end, wavelength_ref=wavelength_midpoint,
+                                              zenith_angle=zenith_angle,
+                                              observatory=observatory, weather=weather)
+                refract_start_pixels = refract_start.asArcseconds()/pixel_scale.asArcseconds()
+                refract_end_pixels = refract_end.asArcseconds()/pixel_scale.asArcseconds()
+                dx = delta(start=refract_start_pixels*np.sin(rotation_angle.asRadians()),
+                           end=refract_end_pixels*np.sin(rotation_angle.asRadians()))
+                dy = delta(start=refract_start_pixels*np.cos(rotation_angle.asRadians()),
+                           end=refract_end_pixels*np.cos(rotation_angle.asRadians()))
+                yield dcr(dx=dx, dy=dy)
+
+    def create_exposure(self, array, elevation, azimuth, variance=None, era=None, snap=0,
+                        exposureId=0, ra=nanAngle, dec=nanAngle, boresightRotAngle=nanAngle,
+                        weather=lsst_weather, **kwargs):
+        """Convert a numpy array to an LSST exposure, and units of electron counts.
+
+        Parameters
+        ----------
+        array : np.ndarray
+            Numpy array to use as the data for the exposure
```

Comment:
I've expanded the description to state where each plane of the MaskedImage comes from.